/*
 * Api Documentation
 * Api Documentation
 *
 * OpenAPI spec version: 1.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package io.swagger.client.api;

import io.swagger.client.ApiClient;
import io.swagger.client.ApiException;
import io.swagger.client.model.UserDTO;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

import java.util.logging.Logger;

/**
 * API tests for AuthControllerApi
 */
public class AuthControllerApiTest {

    static final Logger logger = Logger.getLogger(AuthControllerApiTest.class.getName());

    private final ApiClient apiClient = new ApiClient();
    private final AuthControllerApi api = new AuthControllerApi(apiClient);

    @Test
    public void testLogin_Nominal() throws Exception {
        // Define Input
        UserDTO requestUser = new UserDTO();
        requestUser.setLogin("jdupont");
        requestUser.setPassword("jdupont");

        // Call API
        UserDTO response = null;
        try {
            response = api.loginUsingPOST(requestUser);
        } catch (ApiException e) {
            fail("API Exception :" + e.getCode());
        }

        assertEquals(1, response.getId());
    }

    @Test
    public void testLogin_Unauthorized() throws Exception {
        // Define Input
        UserDTO requestUser = new UserDTO();
        requestUser.setLogin("jdupont");
        requestUser.setPassword("wrongPassword");

        // Call API
        try {
            api.loginUsingPOST(requestUser);
            fail("API Exception expected");
        } catch (ApiException e) {
            assertEquals(401, e.getCode());
        }
    }
    
    @Test
    public void testLogout_Nominal() throws Exception {

        testLogin_Nominal();

        // Call API
        String response = null;
        try {
            response = api.logoutUserUsingPOST();
        } catch (ApiException e) {
            fail("API Exception :" + e.getCode());
        }
        
        assertEquals("You've been signed out!", response);
    }

    @Test
    public void testLogout_NotConnected() throws Exception {

        // Call API
        try {
            api.logoutUserUsingPOST();
            fail("API Exception expected");
        } catch (ApiException e) {
            assertEquals(401, e.getCode());
        }
    }
    
}
